test/common/policy_utils.go

```diff
@@ -204,6 +213,25 @@ func GetLatestStatusMessage(policyName string, templateIdx int) func() string {
 	}
 }
 
+func DoHistoryUpdatedTest(policyName string, messages ...string) {
+	By("Getting policy history")
+	Eventually(func(g Gomega) {
+		history, _, err := GetHistoryMessages(policyName, 0)
+		g.Expect(err).Should(BeNil())
+		for i, msg := range messages {
+			By("Getting history message from history array")
+			historyItem, ok := history[i].(map[string]interface{})
+			g.Expect(ok).Should(BeTrue())
+			order := i + 1
+			By("Checking message is same as [" + msg + "]" + " order number is " + fmt.Sprintf("%d", order))
+			message, _, _ := unstructured.NestedString(historyItem, "message")
+			fmt.Println(msg)
+			fmt.Println(message)
+			g.Expect(message).Should(Equal(msg))
+		}
```

Comment:
I think if the number of messages in the history doesn't match the wanted number of messages, this could run into some trouble. Something might panic, or some messages won't get checked.

Instead of checking them one by one, if this turned the `history` into a simple list of messages, then I think the check could just be something like `g.Expect(historyMessages).Should(Equal(messages))`. And if Gomega does the comparison itself, then I think it will give better debugging information about what items don't match when it fails.

Reply:
how we know which history message broken?